src/m365/purview/commands/retentionlabel/retentionlabel-add.ts

```diff
@@ -0,0 +1,187 @@
+        if (isNaN(args.options.retentionDuration)) {
+          return `Specified retentionDuration ${args.options.retentionDuration} is not a number`;
+        }
+
+        if (args.options.behaviorDuringRetentionPeriod !== 'doNotRetain' &&
+          args.options.behaviorDuringRetentionPeriod !== 'retain' &&
+          args.options.behaviorDuringRetentionPeriod !== 'retainAsRecord' &&
+          args.options.behaviorDuringRetentionPeriod !== 'retainAsRegulatoryRecord') {
+          return `${args.options.behaviorDuringRetentionPeriod} is not a valid behavior of a document with the label. Allowed values are doNotRetain|retain|retainAsRecord|retainAsRegulatoryRecord`;
+        }
+
+        if (args.options.actionAfterRetentionPeriod !== 'none' &&
+          args.options.actionAfterRetentionPeriod !== 'delete' &&
+          args.options.actionAfterRetentionPeriod !== 'startDispositionReview') {
+          return `${args.options.actionAfterRetentionPeriod} is not a valid action to take on a document with the label. Allowed values are none|delete|startDispositionReview`;
+        }
+
+        if (args.options.retentionTrigger &&
+          args.options.retentionTrigger !== 'dateLabeled' &&
+          args.options.retentionTrigger !== 'dateCreated' &&
+          args.options.retentionTrigger !== 'dateModified' &&
+          args.options.retentionTrigger !== 'dateOfEvent') {
+          return `${args.options.retentionTrigger} is not a valid action retention duration calculation. Allowed values are dateLabeled|dateCreated|dateModified|dateOfEvent`;
+        }
+
+        if (args.options.defaultRecordBehavior &&
+          args.options.defaultRecordBehavior !== 'startLocked' &&
+          args.options.defaultRecordBehavior !== 'startUnlocked') {
+          return `${args.options.defaultRecordBehavior} is not a valid state of a record label. Allowed values are startLocked|startUnlocked`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const retentionTrigger: string = args.options.retentionTrigger ? args.options.retentionTrigger : 'dateLabeled';
+    const defaultRecordBehavior: string = args.options.defaultRecordBehavior ? args.options.defaultRecordBehavior : 'startLocked';
+
+    const requestBody: any = {
+      displayName: args.options.displayName,
+      behaviorDuringRetentionPeriod: args.options.behaviorDuringRetentionPeriod,
+      actionAfterRetentionPeriod: args.options.actionAfterRetentionPeriod,
+      retentionTrigger: retentionTrigger,
+      retentionDuration: {
+        '@odata.type': '#microsoft.graph.security.retentionDurationInDays',
+        days: args.options.retentionDuration
+      },
+      defaultRecordBehavior: defaultRecordBehavior
+    };
+
+    if (args.options.descriptionForAdmins) {
+      if (this.verbose) {
+        logger.logToStderr(`Using '${args.options.descriptionForAdmins}' as descriptionForAdmins...`);
```

Comment:
If I am not mistaken the `...` at the end is used to indicate that some process is going on but here we just log information so we may avoid it 🤔

```suggestion
        logger.logToStderr(`Using '${args.options.descriptionForAdmins}' as descriptionForAdmins`);
```

Also similar in the logging below this one